src/hotspot/share/opto/library_call.cpp

```diff
@@ -2031,8 +2031,12 @@ void LibraryCallKit::inline_math_mathExact(Node* math, Node *test) {
     set_control(slow_path);
     set_i_o(i_o());
 
-    uncommon_trap(Deoptimization::Reason_intrinsic,
-                  Deoptimization::Action_none);
+    if (use_builtin_throw) {
+      builtin_throw(Deoptimization::Reason_intrinsic, env()->ArithmeticException_instance());
```

Comment:
I suggest to unconditionally call `builtin_throw()`. It should handle `uncommon_trap` case as well.

What makes sense is to ensure that `builtin_throw()` doesn't change deoptimization reason. It can be implemented with an extra argument to new `GraphKit::builtin_throw` overload (e.g., `bool allow_deopt_reason_none`).